src/main/java/net/dv8tion/jda/internal/requests/restaction/interactions/MessageEditCallbackActionImpl.java

```diff
@@ -71,114 +72,32 @@ public MessageEditCallbackActionImpl deadline(long timestamp)
     @Override
     public MessageEditCallbackActionImpl closeResources()
     {
-        return (MessageEditCallbackActionImpl) super.closeResources();
+        builder.closeFiles();
+        return this;
     }
 
     private boolean isEmpty()
     {
-        return content == null && embeds == null && components == null && !isFileUpdate && files.isEmpty();
+        return builder.isEmpty();
     }
 
     @Override
-    protected DataObject toData()
+    protected RequestBody finalizeData()
     {
         DataObject json = DataObject.empty();
         if (isEmpty())
-            return json.put("type", ResponseType.DEFERRED_MESSAGE_UPDATE.getRaw());
+            return getRequestBody(json.put("type", ResponseType.DEFERRED_MESSAGE_UPDATE.getRaw()));
         json.put("type", ResponseType.MESSAGE_UPDATE.getRaw());
-        DataObject data = DataObject.empty();
-        if (content != null)
-            data.put("content", content);
-        if (embeds != null)
-            data.put("embeds", DataArray.fromCollection(embeds));
-        if (components != null)
-            data.put("components", DataArray.fromCollection(components));
-        json.put("data", data);
-        return json;
-    }
-
-    public MessageEditCallbackAction applyMessage(Message message)
-    {
-        this.content = message.getContentRaw();
-        this.embeds = new ArrayList<>(message.getEmbeds());
-        this.components = new ArrayList<>(message.getActionRows());
-        return this;
-    }
-
-    @Nonnull
-    @Override
-    public MessageEditCallbackAction setEmbeds(@Nonnull Collection<? extends MessageEmbed> embeds)
-    {
-        Checks.noneNull(embeds, "MessageEmbed");
-        Checks.check(embeds.size() <= Message.MAX_EMBED_COUNT, "Cannot have more than %d embeds per message!", Message.MAX_EMBED_COUNT);
-        for (MessageEmbed embed : embeds)
+        try (MessageEditData data = builder.build())
         {
-            Checks.check(embed.isSendable(),
-                    "Provided Message contains an empty embed or an embed with a length greater than %d characters, which is the max for bot accounts!",
-                    MessageEmbed.EMBED_MAX_LENGTH_BOT);
+            json.put("data", data);
```

Comment:
`data` here is `MessageEditData`, not `DataObject`. Is there some magic here or should this be calling `toData`?

Reply:
It implements `SerializableData`, which is special cased by `DataObject#put`.